XCC Build isr_tables.c fail

**Describe the bug**

build FAIL:
file : sections.h
```
#define _SW_ISR_TABLE_SECTION_NAME	.gnu.linkonce.sw_isr_table*
```
build PASS
file : sections.h
```
#define _SW_ISR_TABLE_SECTION_NAME	.gnu.linkonce.sw_isr_table
```

**To Reproduce**
Steps to reproduce the behavior:
west build -b qemu_xtensa samples/hello_world


**Logs and console output**
FAILED: zephyr/arch/common/CMakeFiles/isr_tables.dir/isr_tables.c.obj 
C:\usr\xtensa\XtDevTools\install\tools\RI-2020.5-win32\XtensaTools\bin\xt-xcc.exe -DBUILD_VERSION=zephyr-v2.4.0-1113-g0bcd66ac36d8 -DKERNEL -D_FORTIFY_SOURCE=2 
-D__ZEPHYR_SUPERVISOR__ -D__ZEPHYR__=1 -I../kernel/include -I../arch/xtensa/include -I../include -Izephyr/include/generated -I../soc/xtensa/sample_controller -I../soc/xtensa/sample_controller/include -IE:/work/rtos_work/zephyrproject/modules/hal/xtensa/include -IE:/work/rtos_work/zephyrproject/modules/hal/xtensa/zephyr/soc/sample_controller -isystem ../lib/libc/minimal/include -Os -imacros E:/work/rtos_work/zephyrproject/zephyr/build/zephyr/include/generated/autoconf.h -ffreestanding -fno-common -g -imacrosE:/work/rtos_work/zephyrproject/zephyr/include/toolchain/xcc_missing_defs.h -fms-extensions -imacros E:/work/rtos_work/zephyrproject/zephyr/include/toolchain/zephyr_stdint.h -Wall -Wformat -Wformat-security -Wno-format-zero-length -Wno-main -Wno-pointer-sign -Wpointer-arith -Werror=implicit-int -fno-pic -fno-strict-overflow -ffunction-sections -fdata-sections -mlongcalls -std=c99 -MD -MT zephyr/arch/common/CMakeFiles/isr_tables.dir/isr_tables.c.obj -MF zephyr\arch\common\CMakeFiles\isr_tables.dir\isr_tables.c.obj.d -o zephyr/arch/common/CMakeFiles/isr_tables.dir/isr_tables.c.obj -c E:/work/rtos_work/zephyrproject/zephyr/arch/common/isr_tables.c
C:/Users/xxx/AppData/Local/Temp/cc0spin#1bce.a11124: Assembler messages:
C:/Users/xxx/AppData/Local/Temp/cc0spin#1bce.a11124:220: Warning: missing operand; zero assumed
C:/Users/xxx/AppData/Local/Temp/cc0spin#1bce.a11124:220: Error: invalid operands (*UND* and *ABS* sections) for `*'
xt-xcc.exe ERROR:  C:/usr/xtensa/XtDevTools/install/tools/RI-2020.5-win32/XtensaTools/bin/xt-as.exe returned non-zero status 1
[54/66] Building C object zephyr/lib/posix/CMakeFiles/lib__posix.dir/pthread_common.c.obj
ninja: build stopped: subcommand failed.
FATAL ERROR: command exited with status 1: 'C:\Program Files\CMake\bin\cmake.EXE' --build 'E:\work\rtos_work\zephyrproject\zephyr\build'

**Environment (please complete the following information):**
 - OS: Windows
 - Toolchain XCC(Xplorer 8.0.15)

--- a/arch/arc/core/vector_table.ld
+++ b/arch/arc/core/vector_table.ld
@@ -8,4 +8,4 @@
 
 KEEP(*(.exc_vector_table))
 KEEP(*(".exc_vector_table.*"))
-KEEP(*(_IRQ_VECTOR_TABLE_SECTION_NAME))
+KEEP(*(_IRQ_VECTOR_TABLE_SECTION_SYMS))

--- a/arch/arm/core/aarch32/vector_table.ld
+++ b/arch/arm/core/aarch32/vector_table.ld
@@ -32,7 +32,7 @@ _vector_start = .;
 KEEP(*(.exc_vector_table))
 KEEP(*(".exc_vector_table.*"))
 
-KEEP(*(_IRQ_VECTOR_TABLE_SECTION_NAME))
+KEEP(*(_IRQ_VECTOR_TABLE_SECTION_SYMS))
 
 KEEP(*(.vectors))
 

--- a/include/arch/arm/aarch64/scripts/linker.ld
+++ b/include/arch/arm/aarch64/scripts/linker.ld
@@ -171,7 +171,7 @@ SECTIONS
 
     SECTION_PROLOGUE(_RODATA_SECTION_NAME,,)
     {
-        KEEP(*(_IRQ_VECTOR_TABLE_SECTION_NAME))
+        KEEP(*(_IRQ_VECTOR_TABLE_SECTION_SYMS))
 
         KEEP(*(.openocd_dbg))
         KEEP(*(".openocd_dbg.*"))

--- a/include/linker/common-ram.ld
+++ b/include/linker/common-ram.ld
@@ -9,7 +9,7 @@
 		 * instruction.  See include/sw_isr_table.h.
 		 */
 		. = ALIGN(CONFIG_ARCH_SW_ISR_TABLE_ALIGN);
-		*(_SW_ISR_TABLE_SECTION_NAME)
+		*(_SW_ISR_TABLE_SECTION_SYMS)
 	} GROUP_DATA_LINK_IN(RAMABLE_REGION, ROMABLE_REGION)
 #endif
 

--- a/include/linker/common-rom.ld
+++ b/include/linker/common-rom.ld
@@ -26,7 +26,7 @@
 		 * instruction.  See include/sw_isr_table.h.
 		 */
 		. = ALIGN(CONFIG_ARCH_SW_ISR_TABLE_ALIGN);
-		*(_SW_ISR_TABLE_SECTION_NAME)
+		*(_SW_ISR_TABLE_SECTION_SYMS)
 	} GROUP_LINK_IN(ROMABLE_REGION)
 #endif
 

--- a/include/linker/sections.h
+++ b/include/linker/sections.h
@@ -31,8 +31,11 @@
 #define _UNDEFINED_SECTION_NAME undefined
 
 /* Interrupts */
-#define _IRQ_VECTOR_TABLE_SECTION_NAME	.gnu.linkonce.irq_vector_table*
-#define _SW_ISR_TABLE_SECTION_NAME	.gnu.linkonce.sw_isr_table*
+#define _IRQ_VECTOR_TABLE_SECTION_NAME	.gnu.linkonce.irq_vector_table
+#define _IRQ_VECTOR_TABLE_SECTION_SYMS	.gnu.linkonce.irq_vector_table*
+
+#define _SW_ISR_TABLE_SECTION_NAME	.gnu.linkonce.sw_isr_table
+#define _SW_ISR_TABLE_SECTION_SYMS	.gnu.linkonce.sw_isr_table*
 
 /* Architecture-specific sections */
 #if defined(CONFIG_ARM)

